components/tidb_query_vec_executors/src/table_scan_executor.rs

```diff
@@ -289,23 +293,40 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
         value: &[u8],
         columns: &mut LazyBatchColumnVec,
     ) -> Result<()> {
-        use tidb_query_datatype::codec::{datum, table};
+        use tidb_query_datatype::codec::{datum,table};
 
-        check_record_key(&key)?;
+        table::check_record_key(&key)?;
         let columns_len = self.schema.len();
         let mut decoded_columns = 0;
 
+        // NOTE: `primary_column_ids` is mutual exclusive toward `pk_handle`.
```

Comment:
Please turn this note into an assertion or error check.

Reply:
Done